lib/types.js

```diff
@@ -1626,15 +1626,22 @@ function EnumType(schema, opts) {
   if (!Array.isArray(schema.symbols) || !schema.symbols.length) {
     throw new Error(f('invalid enum symbols: %j', schema.symbols));
   }
+
   this.symbols = Object.freeze(schema.symbols.slice());
   this._indices = {};
+  this.defaultValue = undefined;
   this.symbols.forEach(function (symbol, i) {
     if (!isValidName(symbol)) {
       throw new Error(f('invalid %s symbol: %j', this, symbol));
     }
     if (this._indices[symbol] !== undefined) {
       throw new Error(f('duplicate %s symbol: %j', this, symbol));
     }
+
+    if (schema.default && symbol === schema.default) {
```

Comment:
Rather than check here for each symbol, let's assign once just before `Object.freeze` below:

```js
if (schema.default !== undefined && this._indices[schema.default] === undefined) {
  throw new Error(f('invalid default symbol: %j', schema.default));
}
this.defaultSymbol = schema.default;
```

Validating once at type construction time also helps by saving us from having to do it repeatedly later when reading or writing.

Reply:
All Set